### PR TITLE
Add `on_resurrection` event handler

### PR DESCRIPTION
**Pull request by sensorsasha**
### Overview

The main scope of this PR was to add `on_resurrection` event handler so we can log resurrection events using statsd.

```ruby
Sidekiq::Ultimate.setup! do |config|
  config.on_resurrection = ->(queue_name, jobs_count) do
    puts "Resurrected #{jobs_count} jobs from #{queue_name}"
  end
end
```

Since this repo wasn't touched for a while I did some auxiliary work:

- replace travis CI by github actions
- update and fix rubocop
- improve test coverage
- drop ruby support below 2.7
- add docker-compose for redis, use it for tests and github actions
- add comments to clarify some methods in `resurrector.rb`
- extract timeouts into constants
- fix deprecations

### CODE REVIEW

**Comment by stefanl-st** [comment]
@ixti

**Comment by ixti** [comment]
Can you please open PR on top of main branch of https://github.com/ixti/sidekiq-throttled-2?
I don't have any access to this repo as of the January 2023, thus all development was moved to another repo.

File: lib/sidekiq/ultimate/expirable_set.rb [hunk 1]
```diff
@@ -52,7 +52,7 @@ def initialize
       # @return [ExpirableSet] self
       def add(element, ttl:)
         @mon.synchronize do
-          expires_at = Concurrent.monotonic_time + ttl
+          expires_at = Process.clock_gettime(Process::CLOCK_MONOTONIC) + ttl
```
Comment (ixti) [inline lib/sidekiq/ultimate/expirable_set.rb]: Concurrent.monotonic_time does the same, yet looks cleaner. Sidekiq depends on concurrent gem. Why do we change this?

**Comment by ixti** [comment]
I don't have access to this repo as of January 2023, so I have no objections to the changes. I'm also will be more than happy to transfer the ownership of the gem to whoever wants to maintain it.

**Comment by ixti** [comment]
I suggest using Appraisals for multigem testing, it's usually enough and keeps Gemfile cleaner. Also, I would propose to support ruby 2.7+, 2.6 is long unmantained.

File: lib/sidekiq/ultimate/expirable_set.rb (hunk 1 above)
Comment (sensorsasha) [reply lib/sidekiq/ultimate/expirable_set.rb]: Yes, `Concurrent.monotonic_time` is the same indeed. It was suggested by rubocop 😄

**Comment by ixti** [comment]
Transferred gem ownership to @sensorsasha.

File: lib/sidekiq/ultimate/expirable_set.rb (hunk 1 above)
Comment (ixti) [reply lib/sidekiq/ultimate/expirable_set.rb]: LOL

File: Gemfile.redis41 [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+source "https://rubygems.org"
```
Comment (davidbodow-st) [inline Gemfile.redis41]: Maybe these could be different groups in one gemfile? Open to multiple files as well, but figure it's worth considering

File: README.md [hunk 3]
```diff
@@ -106,11 +115,10 @@ push git commits and tags, and push the `.gem` file to [rubygems.org][].
...
-Copyright (c) 2018 SensorTower Inc.<br>
+Copyright (c) 2018-23 SensorTower Inc.<br>
 See [LICENSE.md][] for further details.
 
 
-[travis.ci]: http://travis-ci.org/sensortower/sidekiq-ultimate
 [rubygems.org]: https://rubygems.org
 [LICENSE.md]: https://github.com/sensortower/sidekiq-ultimate/blob/master/LICENSE.txt
-[sidekiq-throttled]: http://travis-ci.org/sensortower/sidekiq-throttled
+[sidekiq-throttled]: https://github.com/ixti/sidekiq-throttled-tmp
```
Comment (davidbodow-st) [inline README.md]: I'd just link to the rubygems page to avoid the tmp renaming that will soon be changed

**Review by davidbodow-st — APPROVED** [review]

**Comment by davidbodow-st** [comment]
Really appreciate all the additional specs and cleanup @sensorsasha

File: README.md (hunk 3 above)
Comment (sensorsasha) [reply README.md]: URL updated, will not be changed anymore.

File: Gemfile.redis41 (hunk 2 above)
Comment (sensorsasha) [reply Gemfile.redis41]: Changes this to use Appraisals.

File: .rubocop.yml [hunk 4]
```diff
@@ -44,3 +60,12 @@ Style/StringLiterals:
 
 Style/YodaCondition:
   Enabled: false
+
+## RSpec ########################################################################
+RSpec/ExampleLength:
+  Max: 7
+  Enabled: false
```
Comment (DenisSychyov) [inline .rubocop.yml:66]: Do we want to enable it?

**Review by DenisSychyov — APPROVED** [review]
Well done, LGTM 

first_place

File: .rubocop.yml (hunk 4 above)
Comment (sensorsasha) [reply .rubocop.yml:66]: No, I updated the config.

**Pull request merged**